### PR TITLE
Add API endpoints for resolver functionality

### PR DESCRIPTION
**Pull request by neubig**
This is a draft of trying to implement the github resolution functionality in `listen.py`.

The main change in this PR is that it creates an API endpoint `resolve_issue()` in `listen.py` for calling the github resolver.
The input to this function is governed by the data model in `openhands/server/data_models/issue_models.py`.

One big question I have is that this endpoint could take 20 minutes to run, which isn't typically a good way to create an endpoint. Do you have a suggestion for the best practice in implementing this?

Any feedback is welcome (cc @rbren)!

NOTE: separately, running linting commands resulted in some unrelated fixing of linting issues...

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:c088a08-nikolaik   --name openhands-app-c088a08   docker.all-hands.dev/all-hands-ai/openhands:c088a08
```

### CODE REVIEW

File: openhands/server/listen.py [hunk 1]
```diff
@@ -5,28 +5,10 @@
 import time
```
Comment (neubig) [inline openhands/server/listen.py:5]: Ruff formatting is failing on `listen.py`. We should run ruff to fix the formatting.

File: openhands/server/listen.py [hunk 2]
```diff
@@ -452,6 +482,138 @@ async def get_security_analyzers():
...
+        # Get GitHub token from environment
+        github_token = os.environ.get('GITHUB_TOKEN')
+        if not github_token:
+            raise ValueError('GITHUB_TOKEN environment variable not set')
+
+        # Get GitHub username from environment
+        github_username = os.environ.get('GITHUB_USERNAME')
+
+        # Process the issue
+        result = await process_single_issue(
```
Comment (neubig) [inline openhands/server/listen.py:602]: `process_single_issue` doesn't return anything. let's change it to return a pydantic class indicating the result of processing the issue, and modify the code here appropriately.

also, process_single_issue isn't async, so we don't need "await" here.

File: openhands/server/listen.py [hunk 3]
```diff
@@ -452,6 +482,138 @@ async def get_security_analyzers():
...
+        # Get LLM config from current session
+        llm_config = config.get_llm_config()
+
+        # Get GitHub token from environment
+        github_token = os.environ.get('GITHUB_TOKEN')
+        if not github_token:
+            raise ValueError('GITHUB_TOKEN environment variable not set')
+
+        # Get GitHub username from environment
+        github_username = os.environ.get('GITHUB_USERNAME')
```
Comment (neubig) [inline openhands/server/listen.py:599]: `github_username` must be set, so if this returns `None` we should throw an error.

File: openhands/server/data_models/resolver.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
+from typing import Any, Literal
```
Comment (neubig) [inline openhands/server/data_models/resolver.py:1]: Actually, I think we can just remove this `data_models/resolver.py` file. We can just call the functions in `send_pull_request.py` and `resolve_issue.py` directly.

**Review by neubig — COMMENTED** [review]
These tests are failing now:

FAILED tests/unit/resolver/test_resolve_issues.py::test_file_instruction_with_repo_instruction
FAILED tests/unit/test_listen.py::test_resolve_issue_endpoint
FAILED tests/unit/test_listen.py::test_send_pull_request_endpoint

**Comment by openhands-agent** [comment]
New OpenHands update

**Comment by neubig** [comment]
I'm going to close because we've refactored "listen.py" and it'd be easier to start from scratch.

**Pull request closed without merging**